### ISSUE TITLE
[BUG] Claims not validated properly

### ISSUE THREAD

**Issue by Forbinn** [GlitchedPolygons/l8w8jwt#45]
`iss`, `sub`, `aud` and `jti` claims are not validated properly as: if the validation string is shorter than the one in those claims and happens to begin with the same value, it will still be counted as valid.

You can see an example code below:

```c
#include <stdio.h>
#include <string.h>
#include <l8w8jwt/encode.h>
#include <l8w8jwt/decode.h>

int main()
{
    char* jwt;
    size_t jwt_length;

    {
        struct l8w8jwt_encoding_params params;
        l8w8jwt_encoding_params_init(&params);

        params.alg = L8W8JWT_ALG_HS256;
        params.iss = "Black Mesa";

        params.secret_key = (unsigned char*)"YoUR sUpEr S3krEt 1337 HMAC kEy HeRE";
        params.secret_key_length = strlen(params.secret_key);

        params.out = &jwt;
        params.out_length = &jwt_length;

        int r = l8w8jwt_encode(&params);

        printf("\n l8w8jwt example HS512 token: %s \n", r == L8W8JWT_SUCCESS ? jwt : " (encoding failure) ");
    }

    {
        struct l8w8jwt_decoding_params params;
        l8w8jwt_decoding_params_init(&params);

        params.alg = L8W8JWT_ALG_HS256;
        params.jwt = jwt;
        params.jwt_length = strlen(jwt);

        params.verification_key = (unsigned char*)"YoUR sUpEr S3krEt 1337 HMAC kEy HeRE";
        params.verification_key_length = strlen(params.verification_key);

        params.validate_iss = "Black";

        enum l8w8jwt_validation_result validation_result;
        int decode_result = l8w8jwt_decode(&params, &validation_result, NULL, NULL);

        printf("\n Example HS512 token validation %s!\n", (decode_result == L8W8JWT_SUCCESS && validation_result == L8W8JWT_VALID) ? "successful" : "failed");
    }

    l8w8jwt_free(jwt);
}
```

This code will print that the token is valid where in fact it is not. This problem is due to the usage of `strncmp` in [decode.c](https://github.com/GlitchedPolygons/l8w8jwt/blob/2.3.2/src/decode.c#L396) without checking for the size of both string before.

A simple fix would be to make sure that both strings have the same size and only then compare them.

**Comment by GlitchedPolygons** [comment]
@Forbinn yep, good one! Thanks for submitting this. Will look into it soon 👍

**Comment by GlitchedPolygons** [comment]
@Forbinn Done. Producing release `2.3.3` now...  👍 thanks again for pointing out this security issue :)

**Comment by GlitchedPolygons** [comment]
Fixed in [release 2.3.3](https://github.com/GlitchedPolygons/l8w8jwt/releases/tag/2.3.3)
